docs/source/json-api/index.rst

```diff
@@ -32,140 +32,123 @@ We welcome feedback about the JSON API on `our issue tracker
-****************
+1. It talks to the Ledger API to get data it needs to operate, for this it needs to may need to provide an *authentication token* to the Ledger API if the Ledger API requires *authentication*.
+2. It handles the requests coming from one or more Parties, for this each party needs to provide an *authorization token* with each request it sends to the JSON API.
 
-Every request requires you to specify a party and some other settings,
-with a JWT token.  Normal HTTP requests pass the token in an
-``Authentication`` header, while WebSocket requests pass the token in a
-subprotocol.
+Authentication
+--------------
 
-In testing environments, you can use https://jwt.io to generate your
+The authentication token is used exclusively for maintaining the internal list of known packages and templates.
+
+.. note:: At no point should an authentication token be provided to an end user, these are for internal use only.
+
+Every authentication token is different and will depend on what your specific ledger or ledger operator requires.
+The JSON API server requires no access to party-specific data, only access to the ledger identity and package services.
+A token issued for the HTTP JSON API server should contain enough claims to contact these two services but no more than that.
+Please refer to your ledger operator's documentation to find out how.
+
+Once you have retrieved your access token, you can provide it to the JSON API by storing it in a file
+and starting ``daml json-api`` with the flag ``--access-token-file /path/to/your/token.file``.
+
+If the token cannot be read from the provided path or the Ledger API reports an authentication error
+(for example due to token expiration), the JSON API will report the error via logging. 
+
+.. note:: If the token file is updated with a new token it will be picked up at the next attempt to send a request. You can use this to handle cases where an old token expires without interrupting your JSON API service.
+
+Authorization
+-------------
+
+Every request from a client to the JSON API requires you to specify a party and some other settings,
+with a JWT token.  Normal HTTP requests pass the token in an ``Authorization`` header, 
+while WebSocket requests pass the token in a subprotocol.
+
+.. note:: While the JSON API receives the token it doesn't handle it itself. Upon receiving a token it will pass it on to the underlying Ledger's AuthService which will then determine if the token is valid and authorized. As such each token format is ledger/implementation-specific and the below example reflects only the format that is used for the DAML Sandbox.
+
+In the DAML Sandbox testing environment, you can use https://jwt.io to generate your
 token.  The default "header" is fine.  Under "Payload", fill in:
 
 .. code-block:: json
 
     {
       "https://daml.com/ledger-api": {
         "ledgerId": "MyLedger",
-        "applicationId": "foobar",
+        "applicationId": "HTTP-JSON-API-Gateway",
```

Comment:
I’m not sure this is an improvement. This suggest that it is important that it has this specific value which is not the case.

Reply:
Does this not depend on `daml json-api --application-id <value>`? Made it this way since `HTTP-JSON-API-Gateway` is the default.

Could also start `daml json-api --application-id foobar` but I dislike showing flags that people often don't need to use.